llvm/lib/SYCL/PrepareSYCLOpt.cpp

```diff
@@ -60,7 +61,7 @@ struct PrepareSYCLOpt : public ModulePass {
         continue;
       }
       // Annotate kernels for HLS backend being able to identify them
-      if (F.getCallingConv() == CallingConv::SPIR_KERNEL) {
+      if (isKernelFunc(&F)) {
```

Comment:
The replacement is not equivalent, as after the first run of this pass, the kernel will no more have the SPIR_KERNEL calling convention in the HLS flow.
I don't know if there is a problem with adding the same attribute twice on the same function. 
If it is not problematic, I think the gain in clarity is worth the useless work.
Maybe using the `sycl-prepare-clearspir` opt can avoid the useless work while preserving clarity ?